tests/perf-test/ReportActionCompose.perf-test.js

```diff
@@ -0,0 +1,150 @@
+    return {
+        ...actualNav,
+        useNavigation: () => ({
+            navigate: jest.fn(),
+            addListener: () => jest.fn(),
+        }),
+        useIsFocused: () => ({
+            navigate: jest.fn(),
+        }),
+    };
+});
+
+jest.mock('../../src/libs/actions/EmojiPickerAction', () => {
+    const actualEmojiPickerAction = jest.requireActual('../../src/libs/actions/EmojiPickerAction');
+    return {
+        ...actualEmojiPickerAction,
+        emojiPickerRef: {
+            current: {
+                isEmojiPickerVisible: false,
+            },
+        },
+        showEmojiPicker: jest.fn(),
+        hideEmojiPicker: jest.fn(),
+        isActive: () => true,
+    };
+});
+
+beforeAll(() =>
+    Onyx.init({
+        keys: ONYXKEYS,
+        safeEvictionKeys: [ONYXKEYS.COLLECTION.REPORT_ACTIONS],
+        registerStorageEventListener: () => {},
+    }),
+);
+
+// Initialize the network key for OfflineWithFeedback
+beforeEach(() => {
+    Onyx.merge(ONYXKEYS.NETWORK, {isOffline: false});
+});
+
+function ReportActionComposeWrapper() {
+    return (
+        <ComposeProviders components={[OnyxProvider, LocaleContextProvider, KeyboardStateProvider, WindowDimensionsProvider]}>
+            <ReportActionCompose
+                onSubmit={() => jest.fn()}
+                reportID="1"
+                disabled={false}
+                report={LHNTestUtils.getFakeReport()}
+            />
+        </ComposeProviders>
+    );
+}
+const mockEvent = {preventDefault: jest.fn()};
+
+test('should render Composer with text input interactions', async () => {
+    const scenario = async () => {
+        // Query for the composer
+        const composer = await screen.findByTestId('composer');
+
+        expect(composer).toBeDefined();
+        fireEvent.changeText(composer, '@test');
+
+        // Query for the suggestions
+        await screen.findByTestId('suggestions');
+
+        // scroll to hide suggestions
+        fireEvent.scroll(composer);
+
+        // press to block suggestions
+        fireEvent.press(composer);
+    };
+
+    return waitForBatchedUpdates().then(() => measurePerformance(<ReportActionComposeWrapper />, {scenario}));
+});
+
+test('should press composer buttons', async () => {
+    const scenario = async () => {
+        // Query for the attachment button
+        const hintAttachmentButtonText = Localize.translateLocal('reportActionCompose.addAction');
+        const attachmentButton = await screen.findByLabelText(hintAttachmentButtonText);
+
+        expect(attachmentButton).toBeDefined();
+        fireEvent.press(attachmentButton, mockEvent);
+
+        // Query for the emoji button
+        const hintEmojiButtonText = Localize.translateLocal('reportActionCompose.emoji');
+        const emojiButton = await screen.findByLabelText(hintEmojiButtonText);
+
+        expect(emojiButton).toBeDefined();
+        fireEvent.press(emojiButton);
+
```

Comment:
What if we split these scenarios per test? We'll be able to spot regressions more selectively, eg only on sending the messages.

Reply:
not sure about this though, what are your thoughts? Maybe there are some cons to that?

Reply:
I think adding a separate test for each scenario makes more sense. It will be easier to spot potential regression.  Changed.